docs/rules/quote-props.md

```diff
@@ -216,9 +216,35 @@ and the following patterns _stop_ being problems:
 ```js
 var x = {
     "100": 1
+};
+```
+
+A `computed` flag, with default value `false`, can also be used as a modifier for the `"as-needed"` mode. When it is set to `true`, unnecessary property name computations will be reported.
+
+```json
+{
+    "quote-props": [2, "as-needed", {"computed": true}]
 }
 ```
 
+When `computed` is set to `true`, the following patterns become problems:
+
+```js
+/*eslint quote-props: [2, "as-needed", { "computed": true }]*/
+
+var x = {
+    [100]: 1
```

Comment:
May be good to mention the interaction of this with the `numbers` option (this becomes a problem only if `numbers` is set to `false`)